contracts/trees/MerkleTreeWithHistoryPoseidon.sol

```diff
@@ -86,7 +90,7 @@ abstract contract MerkleTreeWithHistoryPoseidon is Initializable {
         
         uint32 newRootIndex = (currentRootIndex + 1) % ROOT_HISTORY_SIZE;
         currentRootIndex = newRootIndex;
-        roots[newRootIndex] = currentLevelHash;
+        roots[newRootIndex] = Root(currentLevelHash, currentRootIndex);
```

Comment:
This is wrong no? `currentRootIndex` is not the `latestLeafIndex` since this value wraps around.

Reply:
What you're looking for is `nextIndex + 2`.